data_steward/tools/create_tier.py

```diff
@@ -0,0 +1,130 @@
+# Python imports
+import argparse
+import logging
+import re
+
+# Third party imports
+
+# Project imports
+
+LOGGER = logging.getLogger(__name__)
+
+
+def validate_tier_param(tier):
+    """
+    helper function to validate the tier parameter passed is either 'controlled' or 'registered'
+
+    :param tier: tier parameter passed through from either a list or command line argument
+    :return: nothing, breaks if not valid
+    """
+    if tier.lower() not in ['controlled', 'registered']:
+        LOGGER.error(
+            f"Parameter ERROR: {tier} is an incorrect input for the tier parameter, accepted: controlled or "
+            f"registered")
+
+
+def validate_deid_stage_param(deid_stage):
+    """
+    helper function to validate the deid_stage parameter passed is correct, must be 'deid', 'base' or 'clean'
+
+    :param deid_stage: deid_stage parameter passed through from either a list or command line argument
+    :return: nothing, breaks if not valid
+    """
+    if deid_stage not in ['deid', 'base', 'clean']:
+        LOGGER.error(
+            f"Parameter ERROR: {deid_stage} is an incorrect input for the deid_stage parameter, accepted: "
+            f"deid, base, clean")
+
+
+def valid_release_tag(arg_value):
+    """
+    User defined helper function to validate that the release_tag parameter follows the correct naming convention
+
+    :param arg_value: release tag parameter passed through either the command line arguments
+    :return: arg_value
+    """
+
+    release_tag_regex = re.compile(r'[0-9]{4}q[0-9]r[0-9]')
+    if not re.match(release_tag_regex, arg_value):
+        LOGGER.error(
+            f"Parameter ERROR {arg_value} is in an incorrect format, accepted: YYYYq#r#"
+        )
+        raise argparse.ArgumentTypeError
+    return arg_value
+
+
+def create_tier(credentials_filepath, project_id, tier, input_dataset,
+                release_tag, deid_stage):
+    """
+    This function is hte main entry point for the deid process.
+    It passes the required parameters to the implementing functions.
+
+    :param credentials_filepath: filepath to credentials to access GCP
+    :param project_id: project_id associated with the input dataset
+    :param tier: controlled or registered tier intended for the output dataset
+    :param input_dataset: name of the input dataset
+    :param release_tag: release tag for dataset in the format of YYYYq#r#
+    :param deid_stage: deid stage (deid, base or clean)
+    :return: name of created controlled or registered dataset
+    """
+
+    # validation of params
+    validate_tier_param(tier)
+    validate_deid_stage_param(deid_stage)
```

Comment:
If you are trying to get DC-1290 done too, you should call the `valid_release_tag` function here too.

Reply:
added!